### PR TITLE
update spring native to 0.12.1

### LINKED ISSUE #51: Upgrade to Spring Native 0.12.0
[Release notes for 0.12.0](https://github.com/spring-projects-experimental/spring-native/releases/tag/0.12.0)

This likely requires Spring Boot 2.7.0, which is being worked on in https://github.com/jhipster/generator-jhipster/pull/18511.

### PR DESCRIPTION
**Pull request by atomfrede**
Seems like we have renamed some cypress files I was not aware of. Will have a look later on the train (if everything works well this time)

closes #51

### CODE REVIEW

File: .github/workflows/jdl.yml [hunk 1]
```diff
@@ -98,7 +98,7 @@ jobs:
       - name: 'SETUP: setup graalvm'
         uses: graalvm/setup-graalvm@v1
         with:
-          version: '22.0.0.2'
+          version: '22.1.0'
```
Comment (mraible) [inline .github/workflows/jdl.yml:101]: 22.2.0 is out now. https://github.com/graalvm/graalvm-ce-builds/releases/tag/vm-22.2.0

**Comment by atomfrede** [comment]
Looks like we are not to far away (mvc + maven/gradle + ubuntu) seems to work. Will try to check the other configs.

**Comment by atomfrede** [comment]
I did not forget about this @mraible Just updated my local graal vm and trying to make it work on my local machine

**Comment by atomfrede** [comment]
I am investigating the webflux issue. Seems to be related to some issues with netty's native image configuration. I have come across [this issue](https://github.com/netty/netty/issues/11369) and [this one](https://github.com/spring-projects-experimental/spring-native/issues/1610), which seems to be exactly our error message.

I have 

* remove the boring ssl dependency --> native build works
* doing "initialize-at-runtime" workaround --> native build works 

Right trying to start the image with updated dependencies. Got some issue with liquibase, but seems to be just configuration.

Will keep you updated with my findings.

**Comment by atomfrede** [comment]
The image starts :D I think the maven webflux part is failing because the replacement/removal of netty-tcnative-boringssl-static is not working as expected.

**Comment by atomfrede** [comment]
Macos and windows builds are failing because testcontainers can't find a docker environment. Not sure why, will have a look.

**Comment by mshima** [comment]
Need to switch windows to use dev profile.

**Comment by atomfrede** [comment]
@mshima You mean changing ` os: [ubuntu-20.04, macos-11, windows-2019]` e.g. to windows-2022?

**Comment by mshima** [comment]
Docker at windows (GitHub Actions) is not compatible with test-containers

**Comment by mshima** [comment]
See https://github.com/hipster-labs/jhipster-daily-builds/commit/e01216926f17a146600049e1ea2c37c08d55d84c.

**Comment by mshima** [comment]
And even cypress is disabled because the applications cannot be started at docker.
https://github.com/jhipster/generator-jhipster-native/blob/03dbdf59bebf907ae1c94e0b53a70f500a8cc977/.github/workflows/jdl.yml#L72

**Comment by atomfrede** [comment]
Thanks @mshima

**Comment by atomfrede** [comment]
So maybe also use the dev profile for macos build

**Comment by mshima** [comment]
MacOS may be related to dependencies https://github.com/testcontainers/testcontainers-java/issues/4008.

**Comment by atomfrede** [comment]
> MacOS may be related to dependencies [testcontainers/testcontainers-java#4008](https://github.com/testcontainers/testcontainers-java/issues/4008).

Thats strange, I use testcontainers nearly on a daily basis and have no issues (arm mac) with it.

**Comment by atomfrede** [comment]
Now this https://github.com/jhipster/generator-jhipster/issues/19225 kicks in

**Comment by mshima** [comment]
> Now this [jhipster/generator-jhipster#19225](https://github.com/jhipster/generator-jhipster/issues/19225) kicks in

It's a different issue.
Native blueprint was using bash before the main generator https://github.com/jhipster/generator-jhipster-native/blob/main/generators/server/templates/npmrc.

The MacOS can be:
https://github.com/testcontainers/testcontainers-java/issues/5030#issuecomment-1033514495

**Comment by mshima** [comment]
The log `com.github.dockerjava` can be enabled to debug the docker problem.

File: .yo-rc.json [hunk 2]
```diff
@@ -16,7 +16,7 @@
         "written": true
       }
     },
-    "jhipsterVersion": "7.6.0",
+    "jhipsterVersion": "7.9.0",
```
Comment (mraible) [inline .yo-rc.json:19]:
```suggestion
    "jhipsterVersion": "7.9.2",
```

**Comment by atomfrede** [comment]
Getting closer. But webflux on MacOS is working 'cause that's what I tried yesterday manually. Will try to have look this evening at the ci.

**Review by mshima — APPROVED** [review]
We can try reenable docker on macOS later at https://github.com/jhipster/generator-jhipster-native/pull/53.

**Comment by atomfrede** [comment]
@mraible @mshima I can't merge it, so feel to merge it.

**Pull request merged**